modules/samtools/collate/main.nf

```diff
@@ -0,0 +1,36 @@
+process SAMTOOLS_COLLATE {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda (params.enable_conda ? "bioconda::samtools=1.15.1" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/samtools:1.15.1--h1170115_0':
+        'quay.io/biocontainers/samtools:1.15.1--h1170115_0' }"
+
+    input:
+    tuple val(meta), path(bam)
```

Comment:
Maybe change `bam` to `input` to clarify that you can also use SAM and CRAM files